lib/minstall.js

```diff
@@ -133,315 +118,586 @@ function checkStartConditions() {
+          continue;
+        }
+
+        for (const locationOfRequestingModule of result[requestedDependencyName][requestedDependencyVersionRange]) {
+          // if the version matches the local module will satisfy the dependency, because even if it would get shadowed,
+          // shadowed dependencies will get fixed with symlinks!
+          const shortenedRequesterLocation = locationOfRequestingModule.substr(cwd.length);
+          const shortenedLocalLocation = localModule.location.substr(cwd.length);
+          logger.debug(`dependency ${requestedDependencyName}@${requestedDependencyVersionRange} requested by '${shortenedRequesterLocation}' will be satisfied by local version ${localModule.version} in '${path.join(shortenedLocalLocation)}'`);
+        }
+        delete result[requestedDependencyName][requestedDependencyVersionRange];
+        break;
+      }
+    }
+    if (Object.keys(result[requestedDependencyName]).length === 0) {
+      delete result[requestedDependencyName];
+    }
   }
 
-  logger.debug('caching conflicting packages:', cachedModuleFolders);
-  const rootNodeModules = path.join(cwd, 'node_modules');
-  const cacheFolder = path.join(rootNodeModules, `${moduleFolderName}_cache`);
-  return systools.moveMultiple(cachedModuleFolders, rootNodeModules, cacheFolder)
-    .then(() => {
-      return cachedModuleFolders;
-    });
+  return result;
 }
 
-function uncacheConflictingModules(cachedModuleFolders, conflictingModules, moduleFolderName) {
+function dependenciesToArray(dependencies) {
+  const result = [];
 
-  if (cachedModuleFolders.length === 0 && conflictingModules.length === 0) {
-    logger.debug('there are no packages that need to be uncached');
-    return Promise.resolve();
+  for (const dependencyName in dependencies) {
+    for (const dependencyVersionRange in dependencies[dependencyName]) {
+      result.push({
+        name: dependencyName,
+        versionRange: dependencyVersionRange,
+        identifier: `${dependencyName}@"${dependencyVersionRange}"`,
+        requestedBy: dependencies[dependencyName][dependencyVersionRange],
+      });
+    }
   }
 
-  const conflictingModuleFolders = conflictingModules.map((module) => {
-    // the conflicting packet was installed into the folderName that it's module-name
-    // requires it to have, thus don't use the realFolderName from the conflicting module
-    // as this might be a local name, that is not the same
-    return module.folderName;
-  });
+  return result;
+}
 
-  logger.debug('uncaching previously cached packages');
-  const rootNodeModules = path.join(cwd, 'node_modules');
-  const cacheFolder = path.join(rootNodeModules, `${moduleFolderName}_cache`);
-  const moduleFolder = path.join(cwd, moduletools.modulesFolder, moduleFolderName);
-  const moduleNodeModules = path.join(moduleFolder, 'node_modules');
+function determineDependencyTargetFolder(requestedDependencyArray, alreadyInstalledDependencies) {
```

Comment:
Inconsistend naming.

Assuming `requestedDependencyArray` and `alreadyInstalledDependencies` are both arrays.